etna/records/views/records.py

```diff
@@ -19,25 +20,30 @@ def record_disambiguation_view(request, reference_number):
 
     https://discovery.nationalarchives.gov.uk/browse/r/h/C4122893
     """
-    pages = Record.search.filter(web_reference=reference_number)
+    per_page = 20
+    page_number = int(request.GET.get("page", 1))
+    offset = (page_number - 1) * per_page
 
-    if len(pages) == 0:
+    count, records = Record.api.search(
+        web_reference=reference_number, offset=offset, size=per_page
+    )
+
+    if len(records) == 0:
         raise Http404
 
     # if the results contain a single record page, redirect to the details page.
-    if len(pages) == 1:
```

Comment:
Maybe a const, "magic number", So maybe __OnlyOneRecodFound

Reply:
Thanks @mozaki1 , I believe this was discussed with @ababic on the call?